opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/SkipPatternTest.java

```diff
@@ -14,21 +10,26 @@
 import org.springframework.boot.test.web.client.TestRestTemplate;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.http.ResponseEntity;
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
-import io.opentracing.mock.MockSpan;
-import io.opentracing.mock.MockTracer;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
 
 /**
  * @author Pavol Loffay
  */
 @SpringBootTest(
         webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
         classes = {SkipPatternTest.SpringConfiguration.class, SkipPatternTest.Controller.class},
-        properties = {"opentracing.spring.web.skipPattern=/skip"})
+        properties = {
+                "opentracing.spring.web.skipPattern=/skip",
+                "opentracing.spring.web.client.enabled=false"
```

Comment:
Is this necessary? if not remove it

Reply:
If it is necessary then the behaviour has changes I guess

Reply:
Yes, it modifies behavior of everything that comes out of `RestTemplateBuilder` provided by Spring Boot. At adds tracing to those instances as well.

Interally, `TestRestTemplate` uses by default this pattern, with private `RestTemplate`.
So, it broke this test, because `TestRestTemplate` produced traces on its own, while test expected exactly one trace from the server.
Another way would be to filter out logged spans, but this was easier and more readable IMO.

This change to `TestRestTemplate` should not break any tests in actual applications.

Reply:
Let's keep this, sleuth also instruments test template

Reply:
Well, that is possible, since you are actually testing spans there as well. Still not typical use case outside OT tests thou.
It is technically correct behavior, since you take TestRestTemplate from Spring context that has tracing enabled, so its logical it traces the requests.. But it is common practice for Spring Boot apps.
I'll try to run tests with snapshot, to see what happens. I'm afraid that it won't be sooner than Monday.

Reply:
Just to close this, I've run tests with ot-spring-cloud and it indeed will need changes. Adding this property to those tests as well will fix it.